library/core/src/ptr/mod.rs

```diff
@@ -767,24 +772,15 @@ pub const unsafe fn read<T>(src: *const T) -> T {
 ///     unaligned: 0x01020304,
 /// };
 ///
-/// #[allow(unaligned_references)]
-/// let v = unsafe {
-///     // Here we attempt to take the address of a 32-bit integer which is not aligned.
-///     let unaligned =
-///         // A temporary unaligned reference is created here which results in
-///         // undefined behavior regardless of whether the reference is used or not.
-///         &packed.unaligned
-///         // Casting to a raw pointer doesn't help; the mistake already happened.
-///         as *const u32;
+/// // Take the address of a 32-bit integer which is not aligned.
+/// // In contrast to `&packed.unaligned as *const _`, this has no undefined behavior.
+/// let unaligned = std::ptr::addr_of!(packed.unaligned);
 ///
-///     let v = std::ptr::read_unaligned(unaligned);
-///
-///     v
-/// };
+/// let v = unsafe { std::ptr::read_unaligned(unaligned) };
+/// assert_eq!({packed.unaligned}, 0x01020304); // `{...}` forces copying the field instead of creating a reference.
```

Comment:
Would it be worth mentioning explicitly that when accessing the field of a packed struct *directly* (i.e., not through a pointer), the compiler knows about it and hence can take care of the alignment issues?

Reply:
Isn't this covered by the line below this change? Or did I misread your suggestion?

Reply:
I have no idea what the "line below this change" is... there seems to be no way to "view surrounding parts of the diff". Man GitHub's UI for these inline comments is bad.^^

Reply:
@RalfJung the way I find it is by shift+click on the file, which opens this part of the diff in a new tab.

Reply:
Ah, usually that gives me some error about commits not being found but it actually worked this time, thanks. :)
However, the line I commented on doesn't even existing any more in the current diff so I think this is irrelevant now.